internal/datastore/postgres/revisions.go

```diff
@@ -91,15 +92,28 @@ func (pgd *pgDatastore) HeadRevision(ctx context.Context) (datastore.Revision, e
 	ctx, span := tracer.Start(ctx, "HeadRevision")
 	defer span.End()
 
+	result, err := pgd.getHeadRevision(ctx, pgd.readPool)
+	if err != nil {
+		return nil, err
+	}
+	if result == nil {
```

Comment:
why not have `getHeadRevision` return datastore.NoRevision here?

Reply:
because the return type is `pgRevision`. I did that so I didn't have to cast from `datastore.Revision` to `pgRevision`.